Themis/Views/Submissions/SubmissionListView.swift

```diff
@@ -33,12 +38,32 @@ struct SubmissionListView: View {
                         Spacer()
                         dateTimeline(submission: submission)
                     }
+                    .swipeActions { cancelButton(for: submission) }
                 }.padding(.trailing)
             }
         }
+        .alert("Are you sure?", isPresented: $presentCancelAlert, presenting: submissionBeingCancelled, actions: { submission in
+            Button("No", role: .cancel, action: {})
+            Button("Yes, cancel", role: .destructive, action: {
+                submissionListVM.cancel(submission, belongingTo: exercise)
+            })
+        }, message: { _ in
+            Text("This will discard the assessment and release the lock on the submission.")
+        })
```

Comment:
```suggestion
        .alert("Are you sure?", isPresented: $presentCancelAlert, presenting: submissionBeingCancelled) { submission in
            Button("No", role: .cancel, action: {})
            Button("Yes, cancel", role: .destructive) {
                submissionListVM.cancel(submission, belongingTo: exercise)
            }
        } message: { _ in
            Text("This will discard the assessment and release the lock on the submission.")
        }
```